### PR TITLE
Fix issue that breaks elodie if coordinate from exiftool is empty

### PR DESCRIPTION
**Pull request by evanjt**
In reference to issue: https://github.com/jmathai/elodie/issues/354

Exiftool returns a string in the case where there is empty data, and python is not happy about casting an empty string to float, so the method here is to first check the length of the variable. If it is 0, then _continue_.

I also made a change that if the variable is not a float, it is then casted to it.

### CODE REVIEW

File: elodie/media/media.py [hunk 1]
```diff
@@ -89,11 +89,20 @@ def get_coordinate(self, type='latitude'):
             if key not in exif:
                 continue
 
+            # Assign coordinate to variable
+            this_coordinate = exif[key]
+
+            # If exiftool GPS output is empty, the data returned will be a str
+            # with 0 length.
+            # https://github.com/jmathai/elodie/issues/354
+            if isinstance(this_coordinate, str) and len(this_coordinate) == 0:
```
Comment (jmathai) [inline elodie/media/media.py]: Can we make this an `elif` under the `if` above?

```
            if key not in exif:
                continue
            elif isinstance(exif[key], str) and len(exif[key]) == 0:
                # ADD COMMENTS HERE
                continue
```

File: elodie/media/media.py [hunk 2]
```diff
@@ -89,11 +89,20 @@ def get_coordinate(self, type='latitude'):
...
+                continue
+
             # Cast coordinate to a float due to a bug in exiftool's
             #   -json output format.
             # https://github.com/jmathai/elodie/issues/171
             # http://u88.n24.queensu.ca/exiftool/forum/index.php/topic,7952.0.html  # noqa
-            this_coordinate = float(exif[key])
+            this_coordinate = float(this_coordinate)
```
Comment (jmathai) [inline elodie/media/media.py]: `float(exif[key])` per change above instead of assigning it and then casting later.

**Comment by evanjt** [comment]
Good suggestions :) made the changes

**Comment by evanjt** [comment]
Made a test for a return of None in the case of an empty GPS field. Also, isinstance() in python2 doesn't compare with str but instead, basestring. Using the six module for compatibility allows the comparison with six.string_types instead.

I've included an image for testing that fails without the bypass.

**Pull request merged**